persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/ClientStateWithPublicAddress.java

```diff
@@ -25,9 +27,18 @@ public ClientStateWithPublicAddress(
     // the remote address is not.
     super(ClientState.forExternalCalls(user, remoteAddress));
     this.publicAddress = publicAddress;
+    this.internal = publicAddress.getAddress().isSiteLocalAddress();
   }
 
   public InetSocketAddress publicAddress() {
     return publicAddress;
   }
+
+  /**
+   * Whether {@link #publicAddress()} is "internal", meaning that the client connected via a local
+   * network rather than the public internet.
+   */
+  public boolean isInternal() {
```

Comment:
There's a bit of a terminology clash here because the class already used "public address", but that address can be public or private now.

Maybe it would be better to rename the class to `ClientStateWithServerAddress`, and this method to `isPrivate()`.